### PR TITLE
added token permissions for testing

### PR DESCRIPTION
**Pull request by jaydee029**
Trying a few changes for PR #2979

added permissions for 
- reusable-call-linter
- reusable-build-on-macos
- reusable-build-on-ubuntu
- reusable-build-on-manylinux

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by jaydee029** [comment]
Hey, @dannypsnl can you please point out , the cause of windows-msvc failing I ddin't changed anything that could directly affect this. (I'm not sure though).

File: .github/workflows/reusable-build-on-macos.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@ on:
 
 jobs:
   build_on_macos:
+    permissions:
+      contents: write
+      packages: write
```
Comment (hydai) [inline .github/workflows/reusable-build-on-macos.yml:19]: We don't use GitHub packages. Is this permission required?

File: .github/workflows/reusable-build-on-manylinux.yml [hunk 2]
```diff
@@ -14,6 +14,9 @@ on:
 
 jobs:
   build_on_manylinux:
+    permissions:
+      contents: write
+      packages: write
```
Comment (hydai) [inline .github/workflows/reusable-build-on-manylinux.yml:19]: We don't use GitHub packages. Is this permission required?

File: .github/workflows/reusable-build-on-ubuntu.yml [hunk 3]
```diff
@@ -14,6 +14,9 @@ on:
 
 jobs:
   build_on_ubuntu:
+    permissions:
+      contents: write
+      packages: write
```
Comment (hydai) [inline .github/workflows/reusable-build-on-ubuntu.yml:19]: We don't use GitHub packages. Is this permission required?

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-build-on-macos.yml (hunk 1 above)
Comment (jaydee029) [reply .github/workflows/reusable-build-on-macos.yml]: Right we don't removing it.

File: .github/workflows/reusable-build-on-manylinux.yml (hunk 2 above)
Comment (jaydee029) [reply .github/workflows/reusable-build-on-manylinux.yml]: Right we don't removing it.

File: .github/workflows/reusable-build-on-ubuntu.yml (hunk 3 above)
Comment (jaydee029) [reply .github/workflows/reusable-build-on-ubuntu.yml]: Right we don't removing it.

**Comment by jaydee029** [comment]
permissions added to 9 more ci workflows

- misc-linters.yml
- build-on-alpine-static
- build-on-android
- build-on-debian-static
- build-on-fedora
- build-on-windows-msvc
- build-on-windows
- create-source-tarball
- static-code-analysis

**Review by hydai — APPROVED** [review]

**Comment by hydai** [comment]
Thanks.

**Pull request merged**